lib/delayed/backend/shared_spec.rb

```diff
@@ -574,7 +574,6 @@ def create_job(opts = {})
       it 're-schedules jobs after failing' do
         worker.work_off
         @job.reload
-        expect(@job.error).to_not be_nil
```

Comment:
`job.error` is assigned on failure to the object being processed, not stored (`last_error` is stored). The back-end object being executed is not `job` declared here, so if you don't have the actual job in hand being worked on there's no way you get the exception back in an attribute.